app/api/src/docker/openscad/openscad.js

```diff
@@ -1,95 +1,88 @@
-  const response = {
-    statusCode: 204,
-    headers: {
-      'Access-Control-Allow-Origin': '*',
-      'Access-Control-Allow-Methods': 'POST',
-      'Access-Control-Allow-Headers': '*',
-    },
-  }
-  callback(null, response)
-}
+const s3 = new AWS.S3()
 
 const preview = async (req, _context, callback) => {
   _context.callbackWaitsForEmptyEventLoop = false
-  const eventBody = Buffer.from(req.body, 'base64').toString('ascii')
-  console.log(eventBody, 'eventBody')
-  const { file, settings } = JSON.parse(eventBody)
-  const { error, result, tempFile } = await runScad({ file, settings })
-  if (error) {
-    const response = {
-      statusCode: 400,
-      body: JSON.stringify({ error, tempFile }),
-    }
-    callback(null, response)
-  } else {
-    console.log(`got result in route: ${result}, file is: ${tempFile}`)
-    const fs = require('fs')
-    const image = fs.readFileSync(`/tmp/${tempFile}/output.png`, {
-      encoding: 'base64',
-    })
-    console.log(image, 'encoded image')
+  const eventBody = req.body
+  console.log('eventBody', eventBody)
+  const key = `${makeHash(eventBody)}.png`
+  console.log('key', key)
+
+  const params = {
+    Bucket: process.env.BUCKET,
+    Key: key,
+  }
+  const previousAsset = await checkIfAlreadyExists(params, s3)
+  if (previousAsset.isAlreadyInBucket) {
+    console.log('already in bucket')
     const response = {
       statusCode: 200,
       body: JSON.stringify({
-        imageBase64: image,
-        result,
-        tempFile,
+        url: getObjectUrl(params, s3),
+        consoleMessage: previousAsset.consoleMessage,
       }),
     }
     callback(null, response)
+    return
   }
-}
 
-const stl = async (req, _context, callback) => {
-  _context.callbackWaitsForEmptyEventLoop = false
-  const eventBody = Buffer.from(req.body, 'base64').toString('ascii')
-  console.log(eventBody, 'eventBody')
-  const { file } = JSON.parse(eventBody)
-  const { error, result, tempFile } = await stlExport({ file })
-  if (error) {
-    const response = {
-      statusCode: 400,
-      body: { error, tempFile },
-    }
-    callback(null, response)
-  } else {
-    console.log(`got result in route: ${result}, file is: ${tempFile}`)
-    const fs = require('fs')
-    const stl = fs.readFileSync(`/tmp/${tempFile}/output.stl`, {
-      encoding: 'base64',
-    })
-    console.log('encoded stl', stl)
-    const response = {
-      statusCode: 200,
-      headers: {
-        'content-type': 'application/stl',
-      },
-      body: stl,
-      isBase64Encoded: true,
-    }
-    console.log('callback fired')
-    callback(null, response)
-  }
+  const { file, settings } = JSON.parse(eventBody)
+  const { error, consoleMessage, fullPath } = await runScad({ file, settings })
+  await storeAssetAndReturnUrl({
+    error,
+    callback,
+    fullPath,
+    consoleMessage,
+    key,
+    s3,
+    params,
+  })
 }
 
+// const stl = async (req, _context, callback) => {
```

Comment:
Should probably just delete this. I should get around to adding stl download soon. Wouldn't be hard.